text/ps-signatures/README.md

```diff
@@ -0,0 +1,144 @@
+    ```rust
+    impl PoKOfSignature {
+        pub fn init(
+            sig: &Signature,
+            vk: &Verkey,
+            messages: &[FieldElement],
+            blindings: Option<&[FieldElement]>,
+            revealed_msg_indices: HashSet<usize>,
+        ) -> Result<Self, PSError>
+    }
+    ```
+- After the challenge is generated (or received in case of an interactive protocol), `PoKOfSignature::gen_proof` is used to create the proof `PoKOfSignatureProof`. 
+    ```rust
+    impl PoKOfSignature {
+        pub fn gen_proof(self, challenge: &FieldElement) -> Result<PoKOfSignatureProof, PSError>
+    }
+    ```
+- The verifier can then verify the proof using `PoKOfSignatureProof::verify`.
+    ```rust
+    impl PoKOfSignatureProof {
+        pub fn verify(
+            &self,
+            vk: &Verkey,
+            revealed_msgs: HashMap<usize, FieldElement>,
+            challenge: &FieldElement,
+        ) -> Result<bool, PSError>
+    }
+    ```
+    Refer the tests [test_PoK_sig](https://github.com/hyperledger/ursa/pull/46/files#diff-d55856d1eb6560e1b74d9b6f2a45acabR252) and [test_PoK_sig_reveal_messages](https://github.com/hyperledger/ursa/pull/46/files#diff-d55856d1eb6560e1b74d9b6f2a45acabR276) for the proof of knowledge of signature and proof of knowledge of signature with revealing some messages respectively.
+- Another test that shows a more elaborate scenario where a signature is requested on a mix of known and committed messages and then the knowledge of that signature is proved in addition to revealing some messages is shown in test [test_scenario_1](https://github.com/hyperledger/ursa/pull/46/files#diff-3b79c2d1f6accb7035421822bd62d6c6R10).
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+The implementation that the RFC is describing is the implementation of sections 6.1 and 6.2 of the [PS signatures paper](https://eprint.iacr.org/2015/525) and an attempt has been made to use the same symbol names in the code as used in the paper. Some implementation optimizations have been done like the use of multi-pairings or multi-exponentiations (both constant and variable time). The PoK protocol for Pedersen commitments used is described in this [RFC](https://github.com/hyperledger/ursa-rfcs/pull/10). PS signatures require type-3 pairings.
+
+The `Signature` implementation is based on section 6.1 of the paper and implements "A Multi-Message Protocol" with the enhancement being the support of signer-known messages also as explained below:
```

Comment:
@hartm @khovratovich @manudrijvers @dubovitskaya This was not mentioned in the paper. Though this looks fine, i still wanted to check with you guys that it is fine. @hartm reviewed it and found it ok.